core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -47,55 +47,79 @@ private[server] class MetadataCache(brokerId: Int) extends Logging {
 
   this.logIdent = "[Kafka Metadata Cache on broker %d] ".format(brokerId)
 
-  def getTopicMetadata(topics: Set[String], protocol: SecurityProtocol) = {
+  private def getAliveEndpoints(brokers: Iterable[Int], protocol: SecurityProtocol): List[Node] = {
+    brokers.flatMap(aliveBrokers.get).toSeq.map(_.getNode(protocol)).toList
```

Comment:
I'd get rid of the `toList` at the end (we have already done a `toSeq` before the `map`) and I'd change the result type to be `Seq[Node]`.